Civi/Payment/PropertyBag.php

```diff
@@ -199,16 +199,21 @@ public static function writeLegacyWarnings() {
 
   /**
    * @param string $prop
+   * @param bool $silent if TRUE return NULL instead of throwing an exception. This is because offsetExists should be safe and not throw exceptions.
    * @return string canonical name.
    * @throws \InvalidArgumentException if prop name not known.
    */
-  protected function handleLegacyPropNames($prop) {
+  protected function handleLegacyPropNames($prop, $silent = FALSE) {
     $newName = static::$propMap[$prop] ?? NULL;
     if ($newName === TRUE) {
       // Good, modern name.
       return $prop;
     }
     if ($newName === NULL) {
+      if ($silent) {
+        // Only for use by offsetExists
+        return;
+      }
```

Comment:
Rather than adding a new "silent" mode here I would suggest `try/catch` in the offsetExists function.

Reply:
That's a good point about performance. I was trying to keep that function simple, but you're right it's internal. Ok, I think this is good.